Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/ChangeEvent.swift

```diff
@@ -0,0 +1,168 @@
+        let docId = document["_id"]
+        return ChangeEvent<Document>.init(
+            id: Document(),
+            operationType: .insert,
+            fullDocument: document,
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+    }
+
+    /**
+     Generates a change event for a local update of a document in the given namespace referring
+     to the given document _id.
+
+     - parameter namespace: the namespace where the document was inserted.
+     - parameter documentId: the _id of the document that was updated.
+     - parameter update: the update specifier.
+     - returns: a change event for a local update of a document in the given namespace referring
+     to the given document _id.
+     */
+    static func changeEventForLocalUpdate(namespace: MongoNamespace,
+                                          documentId: BSONValue,
+                                          update: UpdateDescription,
+                                          fullDocumentAfterUpdate: Document,
+                                          writePending: Bool) -> ChangeEvent<Document> {
+        return ChangeEvent<Document>(
+            id: Document(),
+            operationType: .update,
+            fullDocument: fullDocumentAfterUpdate,
+            ns: namespace,
+            documentKey: ["_id": documentId],
+            updateDescription: update,
+            hasUncommittedWrites: writePending)
+    }
+
+    /**
+     Generates a change event for a local replacement of a document in the given namespace referring
+     to the given document _id.
+
+     - parameter namespace: the namespace where the document was inserted.
+     - parameter documentId: the _id of the document that was updated.
+     - parameter document: the replacement document.
+     - returns: a change event for a local replacement of a document in the given namespace referring
+     to the given document _id.
+     */
+    static func changeEventForLocalReplace(namespace: MongoNamespace,
+                                           documentId: BSONValue,
+                                           document: Document,
+                                           writePending: Bool) -> ChangeEvent<Document> {
+        return ChangeEvent<Document>(
+            id: Document(),
+            operationType: .replace,
+            fullDocument: document,
+            ns: namespace,
+            documentKey: ["_id": documentId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+    }
+
+    /**
+     Generates a change event for a local deletion of a document in the given namespace referring
+     to the given document _id.
+
+     - parameter namespace: the namespace where the document was inserted.
+     - parameter documentId: the _id of the document that was updated.
```

Comment:
[nit] deleted

Reply:
+1, should be [typo] and should be fixed